include/nuttx/list.h

```diff
@@ -37,21 +37,23 @@
 #define LIST_INITIAL_VALUE(list) { &(list), &(list) }
 #define LIST_INITIAL_CLEARED_VALUE { NULL, NULL }
 
-#define list_in_list(item) ((item)->prev || (item)->next)
+#define list_in_list(item) ((item)->prev != NULL)
 #define list_is_empty(list) ((list)->next == list)
```

Comment:
This still is performing a double evaluation, but seems that we do not have other choice

Reply:
Yes, if the macro need return value, it's hard to add temp variable.

Reply:
Now I'm feeling a bit uncomfortable writing this, but still... I re-examined all the source code and didn't find any place except can_receiver that pass a function call as list parameter. So taking into account that we can't fix all the cases I think that it is better:
1. Apply fix in can_receiver
2. Mark PR that change list from inline to macro as breaking change and add a proper description into NuttX documentation.

Sorry again as I do not want to wait anyones time and efforts. I'm fine with current change, just think that above steps are better from user perspective

Reply:
Even if we are not able to eliminate all cases, we should minimize the chance of wrong use by a user. So I still think this PR is a better solution, even if it is not perfect